From rmk+gregkh=suse.de@arm.linux.org.uk  Tue Jan 17 16:39:01 2006
Date: Sat, 14 Jan 2006 20:01:02 +0000
From: Russell King <rmk@arm.linux.org.uk>
To: Greg KH <gregkh@suse.de>
Subject: Fix compiler warning in driver core for CONFIG_HOTPLUG=N
Message-ID: <20060114200101.GD24816@flint.arm.linux.org.uk>
Content-Disposition: inline

FYI, while running a build test, I found:

drivers/base/bus.c:166: warning: `driver_attr_unbind' defined but not used
drivers/base/bus.c:194: warning: `driver_attr_bind' defined but not used

Looks like these two attributes and supporting functions want to be
#ifdef HOTPLUG'd

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/bus.c |    3 +++
 1 file changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/base/bus.c
+++ gregkh-2.6/drivers/base/bus.c
@@ -133,6 +133,8 @@ static struct kobj_type ktype_bus = {
 decl_subsys(bus, &ktype_bus, NULL);
 
 
+#ifdef CONFIG_HOTPLUG
+
 /* Manually detach a device from its associated driver. */
 static int driver_helper(struct device *dev, void *data)
 {
@@ -193,6 +195,7 @@ static ssize_t driver_bind(struct device
 }
 static DRIVER_ATTR(bind, S_IWUSR, NULL, driver_bind);
 
+#endif
 
 static struct device * next_device(struct klist_iter * i)
 {
